### PR TITLE
Cleanup and fix tests

### PR DESCRIPTION
**Pull request by nicoleepp**
Cleans up and adds some documentation to make for easier testing.

Among the changes are:

- Removes the check for "/Users/jacobquinn"
- Introduces an environment variable to specify which database backends to run the tests for
- Updates and moves the sql server tests into the default test script
- Runs the postgresql tests on travis too along with the mysql ones
- Added documentation for docker containers that can be used to test against.
- Deletes unused files in the test directory

### CODE REVIEW

**Comment by nicoleepp** [comment]
This also has changes that address these 2 issues I came across: #187 and #186.

**Comment by omus** [comment]
bump

File: src/Source.jl [hunk 1]
```diff
@@ -139,6 +139,8 @@ function Source(dsn::DSN, query::AbstractString; weakrefstrings::Bool=true, noqu
         allocsize = rowset = rows
     else
         rowset = allocsize = 1
+        # Set rows to 0 because -1 is an invalid number for the Data.Schema() rows parameter
+        rows = 0
```
Comment (quinnj) [inline src/Source.jl]: This should set `rows = missing` to indicate an unknown # of rows.

Comment (quinnj) [reply src/Source.jl]: I went ahead and fixed it.

File: REQUIRE [hunk 2]
```diff
@@ -2,7 +2,7 @@ julia 0.6
 DataStreams 0.2.0
 DataFrames
 CategoricalArrays 0.2
-WeakRefStrings 0.3.0
+WeakRefStrings 0.3.0 0.4.1
```
[the remark this answers is not in the document]
Comment (omus) [reply REQUIRE]: After talking with Nicole there are some incompatibilities with later versions of WeakRefStrings and ODBC. I'll try and make those changes.

**Pull request merged**